osquery/tables/networking/posix/lldp_neighbors.cpp

```diff
@@ -0,0 +1,421 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <iostream>
+#include <map>
+#include <string>
+#include <vector>
+
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include <lldp-const.h>
+#include <lldpctl.h>
+
+namespace osquery {
+namespace tables {
+
+/* kNoAtomStrDefault is the default value used when lldp_atom_get_str does not
+ * return a valid string value. */
+const std::string kNoAtomStrDefault = "unknown";
+
+struct ChassisCapability {
+  bool available;
+  bool enabled;
+};
+
+inline std::string getAtomStr(lldpctl_atom_t* atom, lldpctl_key_t key) {
+  if (!atom) {
+    return kNoAtomStrDefault;
+  }
+
+  const char* val = lldpctl_atom_get_str(atom, key);
+  if (val == NULL) {
+    return kNoAtomStrDefault;
+  }
+
+  return val;
+}
+
+inline std::string commaDelimitedStr(lldpctl_atom_t* things,
+                                     lldpctl_key_t key) {
+  if (!things) {
+    return "";
+  }
+
+  std::string result;
+  lldpctl_atom_t* each;
+
+  lldpctl_atom_foreach(things, each) {
+    result = result + getAtomStr(each, key) + ",";
+  }
+  // Remove trailing comma.
+  if (result.size() > 0) {
```

Comment:
FYI, `if (result.empty()) {`, not sure if there's really a practical difference.